### PR TITLE
Condition updates to the history panel

### PR DESCRIPTION
**Pull request by pmadria3**
This PR updates the condition history panel with a few updates: 

1. Sorts the history panel by date (starting with year, month, then day) and Records with no Date is left aligned
Screen Shot 2022-10-19 at 10 58 55 AM

2. Categories relabeled to Category
Screen Shot 2022-10-19 at 11 27 02 AM

3. takes any date in the yyyyMMDDetc... format and converts into MM/DD/YYYY format. 
Screen Shot 2022-10-19 at 11 35 14 AM
Screen Shot 2022-10-19 at 11 33 43 AM

### CODE REVIEW

File: src/fhir/formatters.ts [hunk 1]
```diff
@@ -34,6 +34,16 @@ export function formatDateLocalToISO(dateStr?: string): string | undefined {
   return date;
 }
 
+// Formats a string yyyymmdd or yyyyMMddHHmmss into MM/DD/YYYY.
+export function formatStringToDate(dateStr?: string): string | undefined {
+  if (!dateStr) return undefined;
+
+  return `${dateStr.substring(4, 6)}/${dateStr.substring(
```
Comment (luthfib) [inline src/fhir/formatters.ts:41]: Sorry having trouble understanding what this is exactly doing, can we leave a comment? or put each part of that string into separate vars... like `${dateStr.substring(4, 6)}` could be a const , etc.

**Review by luthfib — COMMENTED** [review]
Looking, good left a minor comment

File: .changeset/five-peas-live.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (luthfib) [inline .changeset/five-peas-live.md:1]: Maybe this should be patch?

File: src/fhir/formatters.ts [hunk 3]
```diff
@@ -38,10 +38,11 @@ export function formatDateLocalToISO(dateStr?: string): string | undefined {
...
-  return `${dateStr.substring(4, 6)}/${dateStr.substring(
-    6,
-    8
-  )}/${dateStr.substring(0, 4)}`;
+  const month = dateStr.substring(4, 6);
+  const day = dateStr.substring(6, 8);
+  const year = dateStr.substring(0, 4);
+
+  return `${month}/${day}/${year}`;
 }
```
Comment (luthfib) [inline src/fhir/formatters.ts:46]: Wow, much cleaner

File: src/fhir/formatters.ts [hunk 4]
```diff
@@ -34,6 +34,17 @@ export function formatDateLocalToISO(dateStr?: string): string | undefined {
...
 
+// Formats a string yyyymmdd or yyyyMMddHHmmss into MM/DD/YYYY.
+export function formatStringToDate(dateStr?: string): string | undefined {
+  if (!dateStr) return undefined;
+
+  const month = dateStr.substring(4, 6);
+  const day = dateStr.substring(6, 8);
+  const year = dateStr.substring(0, 4);
+
```
Comment (luthfib) [inline src/fhir/formatters.ts:44]: Nice! much cleaner

Comment (luthfib) [inline src/fhir/formatters.ts:44]: Nice! much cleaner

**Review by luthfib — DISMISSED** [review]

**Review by luthfib — APPROVED** [review]

File: src/fhir/formatters.ts [hunk 5]
```diff
@@ -54,3 +65,10 @@ export function formatPhoneNumber(phoneNumber?: string): string | undefined {
   // This would happen if we didn't have the expected 10 digits.
   return match ? `${match[1]}-${match[2]}-${match[3]}` : phoneNumber;
 }
+
+// Formats the date from MM/DD/YYYY to yyyyMMDD.
+export function convertDatetoString(date?: string): string | undefined {
```
Comment (WesleyKapow) [inline src/fhir/formatters.ts:70]: Name is vague. The date is already a string.

You are effectively doing the following `convertDatetoString(formatDateISOToLocal(this.resource.recordedDate))`.

Probably just `resource.recordedDate` would suffice. So remove this function and change the `orderBy` to `c.resource.recordedDate`.

**Pull request merged**